internal/pkg/server/fleet.go

```diff
@@ -458,15 +458,18 @@ func (f *Fleet) runSubsystems(ctx context.Context, cfg *config.Config, g *errgro
 	g.Go(loggedRunFunc(ctx, "Policy monitor", pm.Run))
 
 	// Policy self monitor
-	sm := policy.NewSelfMonitor(cfg.Fleet, bulker, pim, cfg.Inputs[0].Policy.ID, f.reporter)
+	var sm policy.SelfMonitor
 	var agent *model.Agent
-	if f.standAlone {
-		agent, err = f.standAloneSetup(ctx, bulker, sm, cfg.Inputs[0].Policy.ID, cfg.Fleet.Agent.ID)
-		if err != nil {
-			return err
+	if cfg.Fleet.Agent.Checkin {
+		sm := policy.NewSelfMonitor(cfg.Fleet, bulker, pim, cfg.Inputs[0].Policy.ID, f.reporter)
+		if f.standAlone {
```

Comment:
I am assuming by now that "check in" and "standalone" mode are two different things, but I wonder if there is any use case for running standalone with check in (maybe only for development?), or to run in agent without checkin.

Reply:
More about this in https://github.com/elastic/fleet-server/pull/2359#issuecomment-1438380909

Reply:
I do not think we need to support checkin at all in standalone, but for these we probably need to make a change in Kibana to remove the restrictions of having an healthy fleet server agent, I can make a PR that introduce that change with a feature flag if you need it for testing.

Reply:
I agree with @nchaulet, registration + checkin was added just to get around the kibana restrictions.
If we wanted to release a true "stand-alone" server it should not even need to enroll itself

Reply:
We can then fully remove `standAloneSetup`?

Reply:
> I can make a PR that introduce that change with a feature flag if you need it for testing.

@nchaulet that would be great. I guess this would solve part of the _"Add Fleet Server" UI Components_ in https://github.com/elastic/ingest-dev/issues/1530. Instead of using a feature flag we could maybe check if Kibana is runing in standalone mode, if this check is already available.